adapters/webhdfs-adapter/src/main/java/com/connexta/replication/adapters/webhdfs/WebHdfsNodeAdapter.java

```diff
@@ -338,24 +346,50 @@ private String getVersion4Uuid(String fileUrl, Date modificationTime) {
 
   /**
    * Returns the files meeting the criteria for replication by removing elements that: 1) are of
-   * type DIRECTORY or 2) have a modification time before or equal to the filter date, when the
-   * filter date is specified
+   * type DIRECTORY or 2) are not one of the failed IDs and have a modification time before or equal
+   * to the filter date, when the filter date is specified
    *
    * @param files a {@code List} of all {@link FileStatus} objects returned by the GET request
+   * @param failedItemIds a {@code List} of the IDs that failed to be created
    * @param filterDate specifies a point in time such that only files more recent are included; this
    *     value will be set to {@code null} during the first execution of replication
    * @return a resulting {@code List} of {@link FileStatus} objects meeting the criteria
    */
   @VisibleForTesting
-  List<FileStatus> getRelevantFiles(List<FileStatus> files, @Nullable Date filterDate) {
+  List<FileStatus> getRelevantFiles(
+      List<FileStatus> files, List<String> failedItemIds, @Nullable Date filterDate) {
     files.removeIf(
-        file ->
-            file.isDirectory()
-                || (filterDate != null && !file.getModificationTime().after(filterDate)));
+        file -> {
+          // only need to do the lookup if there are items in the failed ID list
+          boolean isFailedId =
+              failedItemIds.isEmpty() ? false : checkIfFailedId(file, failedItemIds);
```

Comment:
I have a few thoughts on this method and the `checkIfFailedId(...)` method.  There are some optimizations we could make, but I'll leave it up to you if you want to look into those for this pull request.  If you do, let's chat about the various options.